FirebaseTestProject/Views/NewPostForm.swift

```diff
@@ -10,31 +10,71 @@ import SwiftUI
 struct NewPostForm: View {
     @State private var title = ""
     @State private var postContent = ""
+    @State private var imageSourceType: ImagePickerView.SourceType?
+    @State private var image: UIImage?
+    
     @FocusState private var showingKeyboard: Bool
+    
     @StateObject private var submitTask = TaskViewModel()
+    
     @Environment(\.user) private var user
     
     var body: some View {
         Form {
             TextField("Title", text: $title)
                 .focused($showingKeyboard)
-            TextField("Post content", text: $postContent)
+            TextEditor(text: $postContent)
                 .focused($showingKeyboard)
+                .multilineTextAlignment(.leading)
+                .frame(width: 300, height: 300, alignment: .topLeading)
+            uploadedImageOrPlaceholder
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 200, height: 200, alignment: .center)
+            VStack {
+                HStack(spacing: 30) {
+                    Text("Attach Image")
+                    Spacer()
+                    Button {
+                        imageSourceType = .photoLibrary
+                    }  label: {
+                        Label("Choose from Library", systemImage: "photo")
+                    }
+                    Button {
+                        imageSourceType = .camera
+                    }  label: {
+                        Label("Take Photo", systemImage: "camera")
+                    }
+                }
+                .buttonStyle(.borderless)
+                .labelStyle(.iconOnly)
+            }
             Button("Submit", action: submitPost)
         }
         .alert("Cannot Submit Post", isPresented: $submitTask.isError, presenting: submitTask.error) { error in
             Text(error.localizedDescription)
         }
+        .sheet(item: $imageSourceType) {
+            ImagePickerView(sourceType: $0, selection: $image)
+        }
+    }
+    
+    private var uploadedImageOrPlaceholder: Image {
+        if let image = image {
+            return Image(uiImage: image)
+        }
+        return Image(systemName: "photo")
```

Comment:
Super minor style thing, but the last return might be better in an `else` block.

Reply:
Added